src/components/common/mediaPicker/MediaPickerSidebarContainer.js

```diff
@@ -0,0 +1,95 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { injectIntl } from 'react-intl';
+import { connect } from 'react-redux';
+import { initializePreviouslySelectedMedia, clearSelectedMedia, resetMetadataShortlist } from '../../../actions/systemActions';
+
+function composeMediaPickerSidebarContainer() {
+  return (ChildComponent) => {
+    class MediaPickerSidebarContainer extends React.Component {
+      state = {
+        // queryInStore: false,
+      };
+
+      UNSAFE_componentWillReceiveProps(nextProps) {
+        // select the media so we fill the reducer with the previously selected media
+        const { initMedia, handleInitialSelectionOfMedia } = this.props;
+        if (JSON.stringify(initMedia) !== JSON.stringify(nextProps.initMedia)) {
+          if (nextProps.initMedia) { // expects an array of media from caller
+            handleInitialSelectionOfMedia(nextProps.initMedia);
+          }
+        }
+        if ((nextProps.selectedMedia !== this.props.selectedMedia)
+          || (nextProps.selectedMedia && this.props.selectedMedia && nextProps.selectedMedia.length !== this.props.selectedMedia.length)) {
+          // if the results have changed from a keyword entry, we need to update the UI
+          this.shouldComponentUpdate(nextProps);
+        }
+      }
+
+      shouldComponentUpdate(nextProps) {
```

Comment:
This method always returns true. Is that what is intended?

Reply:
I think we want it to specifically render if those props above are different...I'll take it out to see if it still renders appropriately

Reply:
The problem I'm seeing is that it's always returning `true` even when checking the props. Shouldn't one of these return `false`?